qualang_tools/octave_tools/README.md

```diff
@@ -0,0 +1,240 @@
+```
+{'offsets': {'I': I_offset, 'Q': Q_offset},
+ 'correction_matrix': (c00, c01, c10, c11)}
+```
+
+If no calibration parameters are found in the database, the function will either:
+* return the dictionary with empty values (verbose_level 0)
+* return the dictionary with empty values and print a warning in the Python console (verbose_level 1)
+* raise an error and block the execution of the script (verbose_level 2)
+
+
+### Usage example
+ 
+```python
+from qualang_tools.octave_tools import set_correction_parameters
+
+# Open the qmm and qm
+qmm = QuantumMachinesManager()
+qm = qmm.open_qm(config)
+
+# Calibrate the desired set of LO frequency and IF
+qm.octave.set_rf_output_gain("qubit", -7)
+qm.calibrate_element("qubit", {5e9: (50e6,)})
+
+# Set the new frequency with calibrated gain
+qm.octave.set_lo_frequency("qubit", 5e9)
+qm.octave.set_rf_output_gain("qubit", -7)
+# Get the correction parameters corresponding to the desired parameters:
+param_qubit = set_correction_parameters(
+    path_to_database="", config=config, element="qubit", 
+    LO=5e9, IF=50e6, gain=-7, 
+    qm=qm, verbose_level=1)
+```
+
+## set_correction_parameters
+This function will look into the calibration database and update the correction parameters in Python, (offsets and 
+correction matrix) corresponding to a given set of intermediate frequency, Octave LO frequency and gain. 
+
+The goal to give the user the ability to easily update the correction parameters while sweeping the LO frequency
+in Python for instance.
+
+The correction parameters are returned in the format of a dictionary:
+```
+{'offsets': {'I': I_offset, 'Q': Q_offset},
+ 'correction_matrix': (c00, c01, c10, c11)}
+```
+
+If no calibration parameters are found in the database, the function will either:
+* return the dictionary with empty values (verbose_level 0)
+* return the dictionary with empty values and print a warning in the Python console (verbose_level 1)
+* raise an error and block the execution of the script (verbose_level 2)
+
+### Usage example
+
+```python
+from qualang_tools.octave_tools import set_correction_parameters
+
+with program() as LO_sweep_prog:
+    # QUA declarations
+    ...
+    with for_(i, 0, i < len(lo_frequencies) + 1, i + 1):
+        pause()  # This waits until it is resumed from python
+        with for_(n, 0, n < n_avg, n + 1):
+            with for_(*from_array(f, frequencies)):
+                # Update the frequency of the digital oscillator linked to the qubit element
+                update_frequency("qubit", f)
+                # Play the pulse
+                ...
+
+# Open the QMM
+qmm = QuantumMachinesManager()
+# Open the quantum machine
+qm = qmm.open_qm(config)
+# Loop over the LO frequencies
+for i in range(len(lo_frequencies)):  
+    # Set the frequency and gain of the LO source
+    qm.octave.set_lo_frequency("qubit", lo_frequencies[i])
+    qm.octave.set_rf_output_gain("qubit", 0)
```

Comment:
Maybe better not to set the gain every time?

Reply:
Good point, it is not needed anymore